shared_objects/lib2.c

```diff
@@ -42,7 +42,9 @@ struct Car *new_car()
 	uint32_t idx = 0;
 	mem[idx++] = auipcc(cs2, 0);
 	mem[idx++] = clc_128(cs3, cs2, 32);
-	mem[idx++] = cincoffsetimm(ca0, cs2, 32 + 16);
+	mem[idx++] = cincoffsetimm(ca0, cs2, 32 + 16); 
+	// 32 is the size of the stubs and 16 is the size of the function pointers in bytes
```

Comment:
I think the comment should go above the first line referencing `32` and `16`. An obvious question, though: can we make these constants based on some sort of "size of" calculation rather than hard coding them?

Reply:
I would have used size of if I could, but there is no way to express the size of these things or I just don't know about it.

Reply:
OK, a comment explaining that it's a bit difficult to calculate and also what platforms the calculation is likely to be correct on would be cool.

Reply:
This should be better now then?

Reply:
No "also what platforms the calculation is likely to be correct on"?

Reply:
This should also be addressed now.